Fails to build on `x86_64-unknown-netbsd`

```
$ git checkout 5.0.0-pre.11
Previous HEAD position was 43ea645 Add missing comma in "As used by:" list
HEAD is now at 71c57b4 Prepare 5.0.0-pre.11
$ cargo build --target x86_64-unknown-netbsd
Updating crates.io index
   Compiling kqueue-sys v1.0.2
error[E0308]: mismatched types
  --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/kqueue-sys-1.0.2/src/lib.rs:68:20
   |
68 |             udata: ptr::null_mut(),
   |                    ^^^^^^^^^^^^^^^ expected `isize`, found *-ptr
   |
   = note:     expected type `isize`
           found raw pointer `*mut _`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `kqueue-sys`

To learn more, run the command again with --verbose.
```

Spotted in https://github.com/rust-lang/rust/pull/87719#issuecomment-891625996.

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -51,6 +51,30 @@ jobs:
         shell: bash
         run: tests/and-retry --features serde
 
+  bsd:
+    strategy:
+      matrix:
+        target:
+          - x86_64-unknown-netbsd
+          - x86_64-unknown-freebsd
+          - x86_64-apple-darwin
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: setup toolchain
+        # `--no-self-update` is needed due to a permission issue on the GHA env.
+        run: |
+          rustup set profile minimal
+          rustup toolchain install stable --no-self-update
+          rustup target add ${{ matrix.target}}
+
+      - name: check build
+        run: |
+          rustc --version && cargo --version  
+          cargo build --target ${{ matrix.target}}
+
   audit:
     runs-on: ubuntu-latest
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -42,7 +42,7 @@ mio = { version = "0.7.7", features = ["os-ext"], optional = true }
 winapi = { version = "0.3.8", features = ["fileapi", "handleapi", "ioapiset", "minwinbase", "synchapi", "winbase", "winnt"] }
 
 [target.'cfg(any(target_os="freebsd", target_os="openbsd", target_os = "netbsd", target_os = "dragonflybsd"))'.dependencies]
-kqueue = "1.0"
+kqueue = "^1.0.3" # fix for #344
 mio = { version = "0.7.7", features = ["os-ext"] }
 
 [dev-dependencies]

